src/test/java/org/kainos/ea/team2/integration/JobIntegrationTest.java

```diff
@@ -81,13 +113,272 @@ void getJobSpec_shouldReturnStatusCode404WhenDaoReturnsNull() {
                 .get();
 
         // check status code 404 returned
-        Assertions.assertEquals(404,response.getStatus());
+        Assertions.assertEquals(404, response.getStatus());
 
         // check response entity is not null
         Assertions.assertEquals("Job does not exist.", response.readEntity(String.class));
 
     }
 
+    /**
+     * Verify that the createJob method returns the ID of the job created
+     * and correct response.
+     */
+    @Test
+    void createJob_shouldReturnIdOfJob_whenJobCreatedSuccessfully() {
+
+        //Dummy job data to test
+        CreateJob job = new CreateJob(
+                "Driver",
+                "Drives bus",
+                "www.bus.com",
+                1,
+                6
+
+        );
+
+        //call url to create job
+        Response response = APP.client().target("http://localhost:8080/api/create-job")
+                .request()
+                .post(Entity.entity(job, MediaType.APPLICATION_JSON_TYPE));
+
+        int id = response.readEntity(Integer.class);
+        int status = response.getStatus();
+
+        //check generated ID isn't null
+        Assertions.assertNotNull(id);
+        //check generated ID is valid
+        Assertions.assertNotEquals(-1, id);
+        //check correct response
+        Assertions.assertEquals(200, status);
+
+    }
+
+    /**
+     * Verify that the createJob method returns status code
+     * 400 and correct error message in entity
+     * when job name is an empty string.
+     */
+    @Test
+    void createJob_shouldReturn400Error_whenJobNameIsAnEmptyString() {
```

Comment:
you only need 1 integration test to check it returns the correct response code when validation fails. all the other checks are done via unit tests